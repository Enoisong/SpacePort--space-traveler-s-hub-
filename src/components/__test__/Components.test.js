import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../Header';
import '@testing-library/jest-dom/extend-expect';

describe('Header component', () => {
  test('renders logo image', () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    expect(screen.getByAltText('logo')).toMatchSnapshot();
  });

  test('renders header title', () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    expect(screen.getByText("Space Travelers' Hub")).toMatchSnapshot();
  });

  test('renders navigation links', () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );

    expect(screen.getByText('Rockets')).toMatchSnapshot();
    expect(screen.getByText('Missions')).toMatchSnapshot();
    expect(screen.getByText('My Profile')).toMatchSnapshot();
  });

  test('renders active navigation link', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <Header />
      </MemoryRouter>,
    );

    expect(screen.getByText('Rockets')).toMatchSnapshot();
    expect(screen.getByText('Missions')).toMatchSnapshot();
    expect(screen.getByText('My Profile')).toMatchSnapshot();
  });
});
