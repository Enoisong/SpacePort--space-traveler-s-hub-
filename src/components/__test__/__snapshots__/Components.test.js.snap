// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders active navigation link 1`] = `
<a
  activeclassname="active"
  aria-current="page"
  class="link-item1 active"
  href="/"
>
  Rockets
</a>
`;

exports[`Header component renders active navigation link 2`] = `
<a
  activeclassname="active"
  class="link-item2"
  href="/missions"
>
  Missions
</a>
`;

exports[`Header component renders active navigation link 3`] = `
<a
  activeclassname="active"
  class="link-item3"
  href="/profile"
>
  My Profile
</a>
`;

exports[`Header component renders header title 1`] = `
<h1>
  Space Travelers' Hub
</h1>
`;

exports[`Header component renders logo image 1`] = `
<img
  alt="logo"
  class="logo"
  src="logo.png"
/>
`;

exports[`Header component renders navigation links 1`] = `
<a
  activeclassname="active"
  aria-current="page"
  class="link-item1 active"
  href="/"
>
  Rockets
</a>
`;

exports[`Header component renders navigation links 2`] = `
<a
  activeclassname="active"
  class="link-item2"
  href="/missions"
>
  Missions
</a>
`;

exports[`Header component renders navigation links 3`] = `
<a
  activeclassname="active"
  class="link-item3"
  href="/profile"
>
  My Profile
</a>
`;
